src/nl/hannahsten/texifyidea/inspections/latex/LatexLabelBeforeCaptionInspection.kt

```diff
@@ -0,0 +1,13 @@
+package nl.hannahsten.texifyidea.inspections.latex
+
+import nl.hannahsten.texifyidea.inspections.TexifyRegexInspection
+import java.util.regex.Pattern
+
+class LatexLabelBeforeCaptionInspection : TexifyRegexInspection(
+        inspectionDisplayName = "Label before caption",
+        inspectionId = "LabelBeforeCaption",
+        errorMessage = { "A label should come after the caption" },
+        pattern = Pattern.compile("(\\\\label\\{.*})(\\s*)(\\\\caption\\{.*})"),
```

Comment:
Although the regex is fine with me, do you think it would work to check if both are in the same environment, to also catch the case when stuff is between the label and caption? I don't know if it would maybe be too slow for large documents.

Reply:
The regex only allows spacing to be between them right?

Reply:
Yes that's my point: ideally you also detect cases where there are other things than whitespace inbetween